syntax = "proto3";

service UserCall {
  rpc AddUser(AddUserRequest) returns (UserReply){}
  rpc RemoveUser(RemoveUserRequest) returns (UserReply){}
  rpc GetAllClassesOfUser(GetAllClassesOfUserRequest) returns (UserReply){}
  rpc GetAllTeachers(GetAllTeachersRequest) returns (UserReply){}
}

message AddUserRequest {
  string username = 1;
  string first_name = 2;
  string last_name = 3;
  string user_password = 4;
  string email = 5;
  int32 role_id = 6;
  optional int32 grade = 7;
}

message RemoveUserRequest {
  int32 user_id = 1;
}

message GetAllClassesOfUserRequest {
  int32 user_id = 1;
}

message GetAllTeachersRequest {
}

message UserReply {
  optional string message = 1;
  optional string error = 2;
  int32 status_code = 3;
}